shared/subprocess/proc.go

```
@@ -0,0 +1,136 @@
+package subprocess
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+)
+
+type Process struct {
+	pid    int64
+	Name   string   `yaml:"name"`
+	Args   []string `yaml:"args,flow"`
+	Stdin  string   `yaml:"stdin"`
+	Stdout string   `yaml:"stdout"`
+	Stderr string   `yaml:"stderr"`
+}
+
+func (p *Process) Pid() (int64, error) {
+	return p.pid, nil
```

Should probably return an error if not running